### ISSUE TITLE
Point moved with native line numbers and wrapped lines

### ISSUE THREAD

**Issue by noctuid** [company-mode/company-mode#727]
When using native line numbers and there are wrapped lines, I have a problem where the company overlay will move the point downward, resulting in text being input later in the buffer if I continue to type. I haven't been able to replicate this entirely with `emacs -Q`. The following will visually move the point to after the overlay and put the overlay over the typed text:

```
(package-initialize)
(require 'company)
(global-company-mode)
(setq display-line-numbers t)
;; to make testing easier
(setq company-minimum-prefix-length 1)
```
Then type enough to wrap the line, enter a newline, type something like `t`, and the point will appear after the company overlay. I'm not sure what in my config is causing the point to stay in the new location if I keep typing, but I think the company popup displaying over the typed text is a bug too.

**Comment by dgutov** [comment]
Thanks for the report. Here's a patch you can try out:

```diff
diff --git a/company.el b/company.el
index ad35411..352f25f 100644
--- a/company.el
+++ b/company.el
@@ -2601,6 +2601,8 @@ If SHOW-VERSION is non-nil, show the version in the echo area."
     ;; Account for the line continuation column.
     (when (zerop (cadr (window-fringes)))
       (cl-decf ww))
+    (when display-line-numbers
+      (cl-decf ww (+ 2 (line-number-display-width))))
     (unless (or (display-graphic-p)
                 (version< "24.3.1" emacs-version))
       ;; Emacs 24.3 and earlier included margins
```

**Comment by noctuid** [comment]
This does not fix the issue for me unfortunately. I also tested again and found that I can replicate this with `emacs -Q`. For example, if I type "tea" (and wait for the overlay after "t") and there are newlines after the point, the "e" will put the point on the line after the overlay, and then the "a" will self-insert on that line.

**Comment by dgutov** [comment]
Sorry, how do I repro? What mode, what buffer contents? Is that affected by the window size?

> For example, if I type "tea" (and wait for the overlay after "t") and there are newlines after the point

So the buffer is originally empty, with only newlines in it? Do you type on the first line?

**Comment by noctuid** [comment]
I just did this in the scratch buffer but can reproduce regardless of major mode and buffer content. It is dependent on the window size; there need to be enough characters for the previous line to be wrapped. There also need to be newlines after the point. The characters typed don't matters as long as there are completions for the first and second (so that the popup isn't canceled). 

Sorry if this wasn't clear. I made an [asciinema demo](https://asciinema.org/a/ZwKFyJb70WemxD9Hf0Jfs8Bcm).
